pynxtools_xrd/reader.py

```diff
@@ -19,162 +19,115 @@
+        "/ENTRY[entry]/2theta_plot/omega": "/Omega",
+        "/ENTRY[entry]/2theta_plot/omega/@units": "/Omega@units",
+        "/ENTRY[entry]/2theta_plot/chi": "/Chi",
+        "/ENTRY[entry]/2theta_plot/phi": "/Phi",
+        "/ENTRY[entry]/2theta_plot/phi/@units": "/Phi@units",
+        "/ENTRY[entry]/COLLECTION[collection]/measurement_type": "/metadata/measurement_type",
+        "/ENTRY[entry]/COLLECTION[collection]/2theta": {
+            "link": "/ENTRY[entry]/2theta_plot/two_theta"
+        },
+        "/ENTRY[entry]/COLLECTION[collection]/omega": "/Omega",
+        "/ENTRY[entry]/COLLECTION[collection]/count_time": "/countTime",
+        "/ENTRY[entry]/COLLECTION[collection]/count_time/@units": "/countTime@units",
+        "/ENTRY[entry]/INSTRUMENT[instrument]/DETECTOR[detector]/scan_axis": "/metadata/scan_axis",
+        "/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/xray_tube_material": "/metadata/source/anode_material",
+        "/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/k_alpha_one": "/metadata/source/kAlpha1",
+        "/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/k_alpha_one/@units": "/metadata/source/kAlpha1@units",
+        "/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/k_alpha_two": "/metadata/source/kAlpha2",
+        "/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/k_alpha_two/@units": "/metadata/source/kAlpha2@units",
+        "/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/ratio_k_alphatwo_k_alphaone": "/metadata/source/ratioKAlpha2KAlpha1",
+        "/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/ratio_k_alphatwo_k_alphaone/@units": "/metadata/source/ratioKAlpha2KAlpha1@units",
+        "/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/kbeta": "/metadata/source/kBeta",
+        "/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/kbeta": "/kBeta@units",
+        "/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/xray_tube_voltage": "/metadata/source/voltage",
+        "/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/xray_tube_voltage/@units": "/metadata/source/voltage@units",
+        "/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/xray_tube_current": "/metadata/source/current",
+        "/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/xray_tube_current/@units": "/metadata/source/current@units",
+        "/ENTRY[entry]/SAMPLE[sample]/sample_id": "/metadata/sample_id",
+        "/ENTRY[entry]/INSTRUMENT[instrument]/DETECTOR[detector]/scan_mode": "/metadata/scan_type",
+    }
+
+    def convert_quantity_to_value_units(self, data_dict):
+        """
+        In a dict, recursively convert every pint.Quantity into value and @units for template
+
+        Args:
+            data_dict (dict): A nested dictionary containing pint.Quantity and other data.
+        """
+        for k, v in list(data_dict.items()):
```

Comment:
Do we need `list` here?

 ```suggestion
        for k, v in data_dict.items():
```

Reply:
Yes. I believe so. It's been a while since I looked into this. We are modifying the dict and adding new keys in some places. This is a copy to keep our loop set of keys unmodified.

Reply:
I see!